src/sagemaker/jumpstart/curated_hub/curated_hub.py

```diff
@@ -113,3 +146,159 @@ def delete_model(self, model_name: str, model_version: str = "*") -> None:
+                if matched_model.version == version:
+                    pass
+
+                # 2. Invalid model version exists in Hub, pass
+                # This will only happen if something goes wrong in our metadata
+                if matched_model.version > version:
+                    pass
+
+                # 3. Old model version exists in Hub, update
+                if matched_model.version < version:
+                    # Check minSDKVersion against current SDK version, emit log
+                    models_to_sync.append(model)
+
+        # Delete old models?
+
+        # Copy content workflow + `SageMaker:ImportHubContent` for each model-to-sync in parallel
+        tasks: List[futures.Future] = []
+        with futures.ThreadPoolExecutor(
+            max_workers=self._default_thread_pool_size,
+            thread_name_prefix="import-models-to-curated-hub",
+        ) as deploy_executor:
+            for model in models_to_sync:
+                task = deploy_executor.submit(self._sync_public_model_to_hub, model)
+                tasks.append(task)
+
+        # Handle failed imports
+        results = futures.wait(tasks)
+        failed_imports: List[Dict[str, Any]] = []
+        for result in results.done:
+            exception = result.exception()
+            if exception:
+                failed_imports.append(
+                    {
+                        "Exception": exception,
+                        "Traceback": "".join(
+                            traceback.TracebackException.from_exception(exception).format()
+                        ),
+                    }
+                )
+        if failed_imports:
+            raise RuntimeError(
+                f"Failures when importing models to curated hub in parallel: {failed_imports}"
+            )
+
+    def _sync_public_model_to_hub(self, model: Dict[str, str]):
+        """Syncs a public JumpStart model version to the Hub. Runs in parallel."""
+        model_name = model["name"]
+        model_version = model["version"]
+
+        model_specs = verify_model_region_and_return_specs(
+            model_id=model_name,
+            version=model_version,
+            region=self.region,
+            scope=JumpStartScriptScope.INFERENCE,
+            sagemaker_session=self._sagemaker_session,
+        )
+        # TODO: Uncomment and implement
+        # studio_specs = self.fetch_studio_specs(model_id=model_name, version=model_version)
+        studio_specs = {}
+
+        dest_location = S3ObjectLocation(
+            bucket=self.hub_bucket_name, key=f"{model_name}/{model_version}"
+        )
+        # TODO: Validations? HeadBucket?
+
+        file_generator = FileGenerator(self.region, self._s3_client, studio_specs)
+        src_files = file_generator.format(model_specs)
+        dest_files = file_generator.format(dest_location)
+
+        files_to_copy = list(FileSync(src_files, dest_files, dest_location).call())
```

Comment:
this is a weird pattern with `list(FileSync(src_files, dest_files, dest_location).call())`? Can we simplify this?

Reply:
I'm removing the `list` in my next revision

Reply:
but even `call()` sounds weird, can we rename the function to something more descriptive? Maybe `execute`?